snooty/test_parser.py

```diff
@@ -534,6 +534,36 @@ def test_rst_replacement() -> None:
     )
 
 
+def test_labels() -> None:
+    path = ROOT_PATH.joinpath(Path("test.rst"))
+    project_config = ProjectConfig(ROOT_PATH, "", source="./")
+    parser = rstparser.Parser(project_config, JSONVisitor)
+    
+    # test label starting with a number
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
+:ref:`100_stacked_example`
+
+.. _100_stacked_example:
```

Comment:
I forget -- is it possible to test the TargetIdentifier in this node in the parser tests?